### PR TITLE
[DML EP] Fix constant node size

### PR DESCRIPTION
**Pull request by sumitsays**
### Description
DML EP should create constant node for only scalar value instead of the entire 1d tensor.

### CODE REVIEW

File: onnxruntime/core/providers/dml/DmlExecutionProvider/src/GraphDescBuilder.cpp [hunk 1]
```diff
@@ -344,35 +372,33 @@ namespace Dml::GraphDescBuilder
...
-                                graphNodes.push_back(std::move(nodeInfo));
-
-                                DML_INTERMEDIATE_GRAPH_EDGE_DESC edge = {};
-                                edge.FromNodeIndex = static_cast<UINT>(graphNodes.size() - 1);
-                                edge.FromNodeOutputIndex = 0;
-                                edge.ToNodeIndex = mainGraphNodeIndex;
-                                edge.ToNodeInputIndex = operatorGraphInputEdge.ToNodeInputIndex;
-                                graphIntermediateEdges.push_back(edge);
+                                    static_cast<const uint8_t*>(constantInput->GetData()) + GetElementSize(tensorDesc->dataType));
+                                if (tensorData.size() <= c_maxConstNodeDataSize)
```
[the remark this answers is not in the document]
Comment (sumitsays) [reply onnxruntime/core/providers/dml/DmlExecutionProvider/src/GraphDescBuilder.cpp]: Agreed. Removed the arbitrary constant variable and just compared with TotalTensorSizeInBytes.

File: onnxruntime/core/providers/dml/DmlExecutionProvider/src/GraphDescBuilder.cpp [hunk 2]
```diff
@@ -344,35 +372,32 @@ namespace Dml::GraphDescBuilder
...
-                                DML_INTERMEDIATE_GRAPH_EDGE_DESC edge = {};
-                                edge.FromNodeIndex = static_cast<UINT>(graphNodes.size() - 1);
-                                edge.FromNodeOutputIndex = 0;
-                                edge.ToNodeIndex = mainGraphNodeIndex;
-                                edge.ToNodeInputIndex = operatorGraphInputEdge.ToNodeInputIndex;
-                                graphIntermediateEdges.push_back(edge);
+                                    tensorData.begin(),
+                                    static_cast<const uint8_t*>(constantInput->GetData()),
+                                    static_cast<const uint8_t*>(constantInput->GetData()) + GetElementSize(tensorDesc->dataType));
+                                if (tensorData.size() <= tensorDesc.TotalTensorSizeInBytes)
```
Comment (jeffbloo) [inline onnxruntime/core/providers/dml/DmlExecutionProvider/src/GraphDescBuilder.cpp:388]: >.TotalTensorSizeInBytes [](http://example.com/codeflow?start=67&length=23)

This should be "->totalTensorSizeInBytes"

File: onnxruntime/core/providers/dml/DmlExecutionProvider/src/GraphDescBuilder.cpp [hunk 3]
```diff
@@ -344,35 +372,32 @@ namespace Dml::GraphDescBuilder
...
+                                {
```
Comment (jeffbloo) [inline onnxruntime/core/providers/dml/DmlExecutionProvider/src/GraphDescBuilder.cpp:389]: This is enabling the first element of the constant to be used as long as the size of that element is <= the total tensor size.  It should be reversed

File: onnxruntime/core/providers/dml/DmlExecutionProvider/src/GraphDescBuilder.cpp [hunk 4]
```diff
@@ -344,35 +372,32 @@ namespace Dml::GraphDescBuilder
...
-
-                                DML_INTERMEDIATE_GRAPH_EDGE_DESC edge = {};
-                                edge.FromNodeIndex = static_cast<UINT>(graphNodes.size() - 1);
-                                edge.FromNodeOutputIndex = 0;
-                                edge.ToNodeIndex = mainGraphNodeIndex;
-                                edge.ToNodeInputIndex = operatorGraphInputEdge.ToNodeInputIndex;
-                                graphIntermediateEdges.push_back(edge);
+                                    tensorData.begin(),
+                                    static_cast<const uint8_t*>(constantInput->GetData()),
+                                    static_cast<const uint8_t*>(constantInput->GetData()) + GetElementSize(tensorDesc->dataType));
```
Comment (jeffbloo) [inline onnxruntime/core/providers/dml/DmlExecutionProvider/src/GraphDescBuilder.cpp:387]: Please move this inside the size check

File: onnxruntime/core/providers/dml/DmlExecutionProvider/src/GraphDescBuilder.cpp [hunk 5]
```diff
@@ -344,35 +372,32 @@ namespace Dml::GraphDescBuilder
...
-                            uint32_t c_maxConstNodeDataSize = 1024 * 1024;
+                            auto& operatorGraphInputNode = graphNodeCreateInfo.nodesAsOperatorDesc[operatorGraphInputEdge.ToNodeIndex];
+                            std::vector<DmlBufferTensorDesc*> toNodeInputTensorDescs = operatorGraphInputNode->GetInputTensors();
+                            DmlBufferTensorDesc* tensorDesc = toNodeInputTensorDescs[operatorGraphInputEdge.ToNodeInputIndex];
 
                             ComPtr<OnnxTensorWrapper> constantInput = constantCpuGraphInputGetter(arg->Name());
 
-                            if (constantInput && constantInput->GetTensorByteSize() < c_maxConstNodeDataSize)
+                            if (constantInput)
                             {
```
Comment (jeffbloo) [inline onnxruntime/core/providers/dml/DmlExecutionProvider/src/GraphDescBuilder.cpp:382]: The modified logic results in asserts during execution.  Let's discuss separately how this is tested.

**Pull request merged**